src/Control/Monad/Fork/Canceler.purs

```diff
@@ -0,0 +1,37 @@
+{-
+Copyright 2016 SlamData, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Control.Monad.Fork.Canceler where
+
+import Prelude
+
+import Control.Monad.Eff.Exception (Error)
+import Data.Monoid (class Monoid)
+import Data.Newtype (class Newtype)
+
+-- | Semigroup (and monoid) for combining cancel functions produced by `fork`.
+newtype Canceler m = Canceler (Error → m Boolean)
```

Comment:
This doesn't work because the Error is abstract in MonadFork. Should be `Canceler e m`.